// react-touch-drag-slider/dist/lib.es.js
import Ye, { useRef as $, useState as fr, useCallback as $e, useEffect as Ie, useLayoutEffect as dr } from "react";
var re = { exports: {} }, V = {};
/**
 * @license React
 * react-jsx-runtime.production.min.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var Le;
function vr() {
  if (Le)
    return V;
  Le = 1;
  var f = Ye, h = Symbol.for("react.element"), y = Symbol.for("react.fragment"), w = Object.prototype.hasOwnProperty, P = f.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED.ReactCurrentOwner, D = { key: !0, ref: !0, __self: !0, __source: !0 };
  function j(E, d, T) {
    var g, b = {}, O = null, u = null;
    T !== void 0 && (O = "" + T), d.key !== void 0 && (O = "" + d.key), d.ref !== void 0 && (u = d.ref);
    for (g in d)
      w.call(d, g) && !D.hasOwnProperty(g) && (b[g] = d[g]);
    if (E && E.defaultProps)
      for (g in d = E.defaultProps, d)
        b[g] === void 0 && (b[g] = d[g]);
    return { $$typeof: h, type: E, key: O, ref: u, props: b, _owner: P.current };
  }
  return V.Fragment = y, V.jsx = j, V.jsxs = j, V;
}
var q = {};
/**
 * @license React
 * react-jsx-runtime.development.js
 *
 * Copyright (c) Facebook, Inc. and its affiliates.
 *
 * This source code is licensed under the MIT license found in the
 * LICENSE file in the root directory of this source tree.
 */
var We;
function pr() {
  return We || (We = 1, process.env.NODE_ENV !== "production" && function() {
    var f = Ye, h = Symbol.for("react.element"), y = Symbol.for("react.portal"), w = Symbol.for("react.fragment"), P = Symbol.for("react.strict_mode"), D = Symbol.for("react.profiler"), j = Symbol.for("react.provider"), E = Symbol.for("react.context"), d = Symbol.for("react.forward_ref"), T = Symbol.for("react.suspense"), g = Symbol.for("react.suspense_list"), b = Symbol.for("react.memo"), O = Symbol.for("react.lazy"), u = Symbol.for("react.offscreen"), v = Symbol.iterator, B = "@@iterator";
    function k(e) {
      if (e === null || typeof e != "object")
        return null;
      var r = v && e[v] || e[B];
      return typeof r == "function" ? r : null;
    }
    var _ = f.__SECRET_INTERNALS_DO_NOT_USE_OR_YOU_WILL_BE_FIRED;
    function p(e) {
      {
        for (var r = arguments.length, t = new Array(r > 1 ? r - 1 : 0), n = 1; n < r; n++)
          t[n - 1] = arguments[n];
        te("error", e, t);
      }
    }
    function te(e, r, t) {
      {
        var n = _.ReactDebugCurrentFrame, o = n.getStackAddendum();
        o !== "" && (r += "%s", t = t.concat([o]));
        var s = t.map(function(i) {
          return String(i);
        });
        s.unshift("Warning: " + r), Function.prototype.apply.call(console[e], console, s);
      }
    }
    var ne = !1, z = !1, G = !1, J = !1, m = !1, C;
    C = Symbol.for("react.module.reference");
    function S(e) {
      return !!(typeof e == "string" || typeof e == "function" || e === w || e === D || m || e === P || e === T || e === g || J || e === u || ne || z || G || typeof e == "object" && e !== null && (e.$$typeof === O || e.$$typeof === b || e.$$typeof === j || e.$$typeof === E || e.$$typeof === d || e.$$typeof === C || e.getModuleId !== void 0));
    }
    function I(e, r, t) {
      var n = e.displayName;
      if (n)
        return n;
      var o = r.displayName || r.name || "";
      return o !== "" ? t + "(" + o + ")" : t;
    }
    function de(e) {
      return e.displayName || "Context";
    }
    function A(e) {
      if (e == null)
        return null;
      if (typeof e.tag == "number" && p("Received an unexpected object in getComponentNameFromType(). This is likely a bug in React. Please file an issue."), typeof e == "function")
        return e.displayName || e.name || null;
      if (typeof e == "string")
        return e;
      switch (e) {
        case w:
          return "Fragment";
        case y:
          return "Portal";
        case D:
          return "Profiler";
        case P:
          return "StrictMode";
        case T:
          return "Suspense";
        case g:
          return "SuspenseList";
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case E:
            var r = e;
            return de(r) + ".Consumer";
          case j:
            var t = e;
            return de(t._context) + ".Provider";
          case d:
            return I(e, e.render, "ForwardRef");
          case b:
            var n = e.displayName || null;
            return n !== null ? n : A(e.type) || "Memo";
          case O: {
            var o = e, s = o._payload, i = o._init;
            try {
              return A(i(s));
            } catch {
              return null;
            }
          }
        }
      return null;
    }
    var L = Object.assign, M = 0, ve, pe, he, ge, me, ye, Ee;
    function be() {
    }
    be.__reactDisabledLog = !0;
    function Ne() {
      {
        if (M === 0) {
          ve = console.log, pe = console.info, he = console.warn, ge = console.error, me = console.group, ye = console.groupCollapsed, Ee = console.groupEnd;
          var e = {
            configurable: !0,
            enumerable: !0,
            value: be,
            writable: !0
          };
          Object.defineProperties(console, {
            info: e,
            log: e,
            warn: e,
            error: e,
            group: e,
            groupCollapsed: e,
            groupEnd: e
          });
        }
        M++;
      }
    }
    function Me() {
      {
        if (M--, M === 0) {
          var e = {
            configurable: !0,
            enumerable: !0,
            writable: !0
          };
          Object.defineProperties(console, {
            log: L({}, e, {
              value: ve
            }),
            info: L({}, e, {
              value: pe
            }),
            warn: L({}, e, {
              value: he
            }),
            error: L({}, e, {
              value: ge
            }),
            group: L({}, e, {
              value: me
            }),
            groupCollapsed: L({}, e, {
              value: ye
            }),
            groupEnd: L({}, e, {
              value: Ee
            })
          });
        }
        M < 0 && p("disabledDepth fell below zero. This is a bug in React. Please file an issue.");
      }
    }
    var ae = _.ReactCurrentDispatcher, ie;
    function H(e, r, t) {
      {
        if (ie === void 0)
          try {
            throw Error();
          } catch (o) {
            var n = o.stack.trim().match(/\n( *(at )?)/);
            ie = n && n[1] || "";
          }
        return `
` + ie + e;
      }
    }
    var oe = !1, X;
    {
      var Ue = typeof WeakMap == "function" ? WeakMap : Map;
      X = new Ue();
    }
    function Re(e, r) {
      if (!e || oe)
        return "";
      {
        var t = X.get(e);
        if (t !== void 0)
          return t;
      }
      var n;
      oe = !0;
      var o = Error.prepareStackTrace;
      Error.prepareStackTrace = void 0;
      var s;
      s = ae.current, ae.current = null, Ne();
      try {
        if (r) {
          var i = function() {
            throw Error();
          };
          if (Object.defineProperty(i.prototype, "props", {
            set: function() {
              throw Error();
            }
          }), typeof Reflect == "object" && Reflect.construct) {
            try {
              Reflect.construct(i, []);
            } catch (F) {
              n = F;
            }
            Reflect.construct(e, [], i);
          } else {
            try {
              i.call();
            } catch (F) {
              n = F;
            }
            e.call(i.prototype);
          }
        } else {
          try {
            throw Error();
          } catch (F) {
            n = F;
          }
          e();
        }
      } catch (F) {
        if (F && n && typeof F.stack == "string") {
          for (var a = F.stack.split(`
`), R = n.stack.split(`
`), c = a.length - 1, l = R.length - 1; c >= 1 && l >= 0 && a[c] !== R[l]; )
            l--;
          for (; c >= 1 && l >= 0; c--, l--)
            if (a[c] !== R[l]) {
              if (c !== 1 || l !== 1)
                do
                  if (c--, l--, l < 0 || a[c] !== R[l]) {
                    var x = `
` + a[c].replace(" at new ", " at ");
                    return e.displayName && x.includes("<anonymous>") && (x = x.replace("<anonymous>", e.displayName)), typeof e == "function" && X.set(e, x), x;
                  }
                while (c >= 1 && l >= 0);
              break;
            }
        }
      } finally {
        oe = !1, ae.current = s, Me(), Error.prepareStackTrace = o;
      }
      var N = e ? e.displayName || e.name : "", Fe = N ? H(N) : "";
      return typeof e == "function" && X.set(e, Fe), Fe;
    }
    function Ve(e, r, t) {
      return Re(e, !1);
    }
    function qe(e) {
      var r = e.prototype;
      return !!(r && r.isReactComponent);
    }
    function K(e, r, t) {
      if (e == null)
        return "";
      if (typeof e == "function")
        return Re(e, qe(e));
      if (typeof e == "string")
        return H(e);
      switch (e) {
        case T:
          return H("Suspense");
        case g:
          return H("SuspenseList");
      }
      if (typeof e == "object")
        switch (e.$$typeof) {
          case d:
            return Ve(e.render);
          case b:
            return K(e.type, r, t);
          case O: {
            var n = e, o = n._payload, s = n._init;
            try {
              return K(s(o), r, t);
            } catch {
            }
          }
        }
      return "";
    }
    var Z = Object.prototype.hasOwnProperty, _e = {}, we = _.ReactDebugCurrentFrame;
    function Q(e) {
      if (e) {
        var r = e._owner, t = K(e.type, e._source, r ? r.type : null);
        we.setExtraStackFrame(t);
      } else
        we.setExtraStackFrame(null);
    }
    function Be(e, r, t, n, o) {
      {
        var s = Function.call.bind(Z);
        for (var i in e)
          if (s(e, i)) {
            var a = void 0;
            try {
              if (typeof e[i] != "function") {
                var R = Error((n || "React class") + ": " + t + " type `" + i + "` is invalid; it must be a function, usually from the `prop-types` package, but received `" + typeof e[i] + "`.This often happens because of typos such as `PropTypes.function` instead of `PropTypes.func`.");
                throw R.name = "Invariant Violation", R;
              }
              a = e[i](r, i, n, t, null, "SECRET_DO_NOT_PASS_THIS_OR_YOU_WILL_BE_FIRED");
            } catch (c) {
              a = c;
            }
            a && !(a instanceof Error) && (Q(o), p("%s: type specification of %s `%s` is invalid; the type checker function must return `null` or an `Error` but returned a %s. You may have forgotten to pass an argument to the type checker creator (arrayOf, instanceOf, objectOf, oneOf, oneOfType, and shape all require an argument).", n || "React class", t, i, typeof a), Q(null)), a instanceof Error && !(a.message in _e) && (_e[a.message] = !0, Q(o), p("Failed %s type: %s", t, a.message), Q(null));
          }
      }
    }
    var ze = Array.isArray;
    function se(e) {
      return ze(e);
    }
    function Ge(e) {
      {
        var r = typeof Symbol == "function" && Symbol.toStringTag, t = r && e[Symbol.toStringTag] || e.constructor.name || "Object";
        return t;
      }
    }
    function Je(e) {
      try {
        return Pe(e), !1;
      } catch {
        return !0;
      }
    }
    function Pe(e) {
      return "" + e;
    }
    function Te(e) {
      if (Je(e))
        return p("The provided key is an unsupported type %s. This value must be coerced to a string before before using it here.", Ge(e)), Pe(e);
    }
    var U = _.ReactCurrentOwner, He = {
      key: !0,
      ref: !0,
      __self: !0,
      __source: !0
    }, xe, Oe, ue;
    ue = {};
    function Xe(e) {
      if (Z.call(e, "ref")) {
        var r = Object.getOwnPropertyDescriptor(e, "ref").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.ref !== void 0;
    }
    function Ke(e) {
      if (Z.call(e, "key")) {
        var r = Object.getOwnPropertyDescriptor(e, "key").get;
        if (r && r.isReactWarning)
          return !1;
      }
      return e.key !== void 0;
    }
    function Ze(e, r) {
      if (typeof e.ref == "string" && U.current && r && U.current.stateNode !== r) {
        var t = A(U.current.type);
        ue[t] || (p('Component "%s" contains the string ref "%s". Support for string refs will be removed in a future major release. This case cannot be automatically converted to an arrow function. We ask you to manually fix this case by using useRef() or createRef() instead. Learn more about using refs safely here: https://reactjs.org/link/strict-mode-string-ref', A(U.current.type), e.ref), ue[t] = !0);
      }
    }
    function Qe(e, r) {
      {
        var t = function() {
          xe || (xe = !0, p("%s: `key` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "key", {
          get: t,
          configurable: !0
        });
      }
    }
    function er(e, r) {
      {
        var t = function() {
          Oe || (Oe = !0, p("%s: `ref` is not a prop. Trying to access it will result in `undefined` being returned. If you need to access the same value within the child component, you should pass it as a different prop. (https://reactjs.org/link/special-props)", r));
        };
        t.isReactWarning = !0, Object.defineProperty(e, "ref", {
          get: t,
          configurable: !0
        });
      }
    }
    var rr = function(e, r, t, n, o, s, i) {
      var a = {
        $$typeof: h,
        type: e,
        key: r,
        ref: t,
        props: i,
        _owner: s
      };
      return a._store = {}, Object.defineProperty(a._store, "validated", {
        configurable: !1,
        enumerable: !1,
        writable: !0,
        value: !1
      }), Object.defineProperty(a, "_self", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: n
      }), Object.defineProperty(a, "_source", {
        configurable: !1,
        enumerable: !1,
        writable: !1,
        value: o
      }), Object.freeze && (Object.freeze(a.props), Object.freeze(a)), a;
    };
    function tr(e, r, t, n, o) {
      {
        var s, i = {}, a = null, R = null;
        t !== void 0 && (Te(t), a = "" + t), Ke(r) && (Te(r.key), a = "" + r.key), Xe(r) && (R = r.ref, Ze(r, o));
        for (s in r)
          Z.call(r, s) && !He.hasOwnProperty(s) && (i[s] = r[s]);
        if (e && e.defaultProps) {
          var c = e.defaultProps;
          for (s in c)
            i[s] === void 0 && (i[s] = c[s]);
        }
        if (a || R) {
          var l = typeof e == "function" ? e.displayName || e.name || "Unknown" : e;
          a && Qe(i, l), R && er(i, l);
        }
        return rr(e, a, R, o, n, U.current, i);
      }
    }
    var ce = _.ReactCurrentOwner, Ce = _.ReactDebugCurrentFrame;
    function Y(e) {
      if (e) {
        var r = e._owner, t = K(e.type, e._source, r ? r.type : null);
        Ce.setExtraStackFrame(t);
      } else
        Ce.setExtraStackFrame(null);
    }
    var le;
    le = !1;
    function fe(e) {
      return typeof e == "object" && e !== null && e.$$typeof === h;
    }
    function Se() {
      {
        if (ce.current) {
          var e = A(ce.current.type);
          if (e)
            return `

Check the render method of \`` + e + "`.";
        }
        return "";
      }
    }
    function nr(e) {
      {
        if (e !== void 0) {
          var r = e.fileName.replace(/^.*[\\\/]/, ""), t = e.lineNumber;
          return `

Check your code at ` + r + ":" + t + ".";
        }
        return "";
      }
    }
    var je = {};
    function ar(e) {
      {
        var r = Se();
        if (!r) {
          var t = typeof e == "string" ? e : e.displayName || e.name;
          t && (r = `

Check the top-level render call using <` + t + ">.");
        }
        return r;
      }
    }
    function ke(e, r) {
      {
        if (!e._store || e._store.validated || e.key != null)
          return;
        e._store.validated = !0;
        var t = ar(r);
        if (je[t])
          return;
        je[t] = !0;
        var n = "";
        e && e._owner && e._owner !== ce.current && (n = " It was passed a child from " + A(e._owner.type) + "."), Y(e), p('Each child in a list should have a unique "key" prop.%s%s See https://reactjs.org/link/warning-keys for more information.', t, n), Y(null);
      }
    }
    function De(e, r) {
      {
        if (typeof e != "object")
          return;
        if (se(e))
          for (var t = 0; t < e.length; t++) {
            var n = e[t];
            fe(n) && ke(n, r);
          }
        else if (fe(e))
          e._store && (e._store.validated = !0);
        else if (e) {
          var o = k(e);
          if (typeof o == "function" && o !== e.entries)
            for (var s = o.call(e), i; !(i = s.next()).done; )
              fe(i.value) && ke(i.value, r);
        }
      }
    }
    function ir(e) {
      {
        var r = e.type;
        if (r == null || typeof r == "string")
          return;
        var t;
        if (typeof r == "function")
          t = r.propTypes;
        else if (typeof r == "object" && (r.$$typeof === d || r.$$typeof === b))
          t = r.propTypes;
        else
          return;
        if (t) {
          var n = A(r);
          Be(t, e.props, "prop", n, e);
        } else if (r.PropTypes !== void 0 && !le) {
          le = !0;
          var o = A(r);
          p("Component %s declared `PropTypes` instead of `propTypes`. Did you misspell the property assignment?", o || "Unknown");
        }
        typeof r.getDefaultProps == "function" && !r.getDefaultProps.isReactClassApproved && p("getDefaultProps is only used on classic React.createClass definitions. Use a static property named `defaultProps` instead.");
      }
    }
    function or(e) {
      {
        for (var r = Object.keys(e.props), t = 0; t < r.length; t++) {
          var n = r[t];
          if (n !== "children" && n !== "key") {
            Y(e), p("Invalid prop `%s` supplied to `React.Fragment`. React.Fragment can only have `key` and `children` props.", n), Y(null);
            break;
          }
        }
        e.ref !== null && (Y(e), p("Invalid attribute `ref` supplied to `React.Fragment`."), Y(null));
      }
    }
    function Ae(e, r, t, n, o, s) {
      {
        var i = S(e);
        if (!i) {
          var a = "";
          (e === void 0 || typeof e == "object" && e !== null && Object.keys(e).length === 0) && (a += " You likely forgot to export your component from the file it's defined in, or you might have mixed up default and named imports.");
          var R = nr(o);
          R ? a += R : a += Se();
          var c;
          e === null ? c = "null" : se(e) ? c = "array" : e !== void 0 && e.$$typeof === h ? (c = "<" + (A(e.type) || "Unknown") + " />", a = " Did you accidentally export a JSX literal instead of a component?") : c = typeof e, p("React.jsx: type is invalid -- expected a string (for built-in components) or a class/function (for composite components) but got: %s.%s", c, a);
        }
        var l = tr(e, r, t, o, s);
        if (l == null)
          return l;
        if (i) {
          var x = r.children;
          if (x !== void 0)
            if (n)
              if (se(x)) {
                for (var N = 0; N < x.length; N++)
                  De(x[N], e);
                Object.freeze && Object.freeze(x);
              } else
                p("React.jsx: Static children should always be an array. You are likely explicitly calling React.jsxs or React.jsxDEV. Use the Babel transform instead.");
            else
              De(x, e);
        }
        return e === w ? or(l) : ir(l), l;
      }
    }
    function sr(e, r, t) {
      return Ae(e, r, t, !0);
    }
    function ur(e, r, t) {
      return Ae(e, r, t, !1);
    }
    var cr = ur, lr = sr;
    q.Fragment = w, q.jsx = cr, q.jsxs = lr;
  }()), q;
}
(function(f) {
  process.env.NODE_ENV === "production" ? f.exports = vr() : f.exports = pr();
})(re);
const hr = re.exports.Fragment, W = re.exports.jsx, gr = re.exports.jsxs, mr = String.raw;
function yr({
  child: f,
  sliderWidth: h,
  sliderHeight: y,
  scaleOnDrag: w = !1
}) {
  const P = $(null), D = () => {
    w && P.current && (P.current.style.transform = "scale(0.9)");
  }, j = () => {
    w && P.current && (P.current.style.transform = "scale(1)");
  };
  return /* @__PURE__ */ gr(hr, {
    children: [/* @__PURE__ */ W("style", {
      children: mr`
          .rtds-single-slide-styles img {
            max-width: 100%;
            max-height: 100%;
          }
        `
    }), /* @__PURE__ */ W("div", {
      ref: P,
      style: {
        width: `${h}px`,
        height: `${y}px`,
        transition: "transform 0.2s ease-out"
      },
      className: "rtds-single-slide-styles",
      children: /* @__PURE__ */ W("div", {
        style: {
          padding: "1rem",
          height: "100%",
          display: "flex",
          alignItems: "center",
          justifyContent: "center",
          userSelect: "none"
        },
        onPointerDown: D,
        onPointerUp: j,
        onPointerLeave: j,
        onDragStart: (E) => (E.preventDefault(), E.stopPropagation(), !1),
        children: f
      })
    })]
  });
}
function ee(f) {
  const h = f.clientWidth, y = f.clientHeight;
  return { width: h, height: y };
}
if (import.meta.vitest) {
  const { it: f, expect: h } = import.meta.vitest, y = document.createElement("div");
  f("Gets an elements dimensions", () => {
    h(ee(y)).toStrictEqual({
      width: 0,
      height: 0
    });
  });
}
function br({
  children: f,
  onSlideComplete: h,
  onSlideStart: y,
  activeIndex: w = null,
  threshHold: P = 100,
  transition: D = 0.3,
  scaleOnDrag: j = !1
}) {
  const [E, d] = fr({
    width: 0,
    height: 0
  }), T = $(!1), g = $(0), b = $(0), O = $(0), u = $(0), v = $(null), B = $(null), k = $e((m = E.width) => {
    b.current = u.current * -m, O.current = b.current, J();
  }, [E.width]), _ = $e(() => {
    v.current && (v.current.style.transition = `transform ${D}s ease-out`);
  }, [D]), p = () => {
    v.current && (v.current.style.transition = "none");
  };
  Ie(() => {
    w !== u.current && (_(), u.current = w, k());
  }, [w, k, _]), dr(() => {
    v.current && (p(), d(ee(v.current)), k(ee(v.current).width));
  }, [k]), Ie(() => {
    const m = () => {
      if (p(), v.current) {
        const {
          width: S,
          height: I
        } = ee(v.current);
        d({
          width: S,
          height: I
        }), k(S);
      }
    }, C = ({
      key: S
    }) => {
      const I = ["ArrowRight", "ArrowLeft"].includes(S);
      I && _(), I && y && y(u.current), S === "ArrowRight" && u.current < f.length - 1 && (u.current += 1), S === "ArrowLeft" && u.current > 0 && (u.current -= 1), I && h && h(u.current), k();
    };
    return window.addEventListener("resize", m), window.addEventListener("keydown", C), () => {
      window.removeEventListener("resize", m), window.removeEventListener("keydown", C);
    };
  }, [f.length, k, h, y, _]);
  function te(m) {
    return function(C) {
      _(), u.current = m, g.current = C.pageX, T.current = !0, B.current = requestAnimationFrame(G), v.current && (v.current.style.cursor = "grabbing"), y && y(u.current);
    };
  }
  function ne(m) {
    if (T.current) {
      const C = m.pageX;
      b.current = O.current + C - g.current;
    }
  }
  function z() {
    _(), cancelAnimationFrame(B.current), T.current = !1;
    const m = b.current - O.current;
    m < -P && u.current < f.length - 1 && (u.current += 1), m > P && u.current > 0 && (u.current -= 1), _(), k(), v.current.style.cursor = "grab", h && h(u.current);
  }
  function G() {
    J(), T.current && requestAnimationFrame(G);
  }
  function J() {
    !v.current || (v.current.style.transform = `translateX(${b.current}px)`);
  }
  return /* @__PURE__ */ W("div", {
    className: "rtds-slider-wrapper",
    style: {
      overflow: "hidden",
      width: "100%",
      height: "100%",
      maxHeight: "100vh"
    },
    children: /* @__PURE__ */ W("div", {
      "data-testid": "slider",
      ref: v,
      className: "rtds-slider-styles",
      style: {
        all: "initial",
        width: "100%",
        height: "100%",
        maxHeight: "100vh",
        display: "inline-flex",
        willChange: "transform, scale",
        cursor: "grab"
      },
      children: f.map((m, C) => /* @__PURE__ */ W("div", {
        onPointerDown: te(C),
        onPointerMove: ne,
        onPointerUp: z,
        onPointerLeave: () => {
          T.current && z();
        },
        onContextMenu: (S) => {
          S.preventDefault(), S.stopPropagation();
        },
        className: "slide-outer",
        style: {
          touchAction: "none"
        },
        children: /* @__PURE__ */ W(yr, {
          child: m,
          sliderWidth: E.width,
          sliderHeight: E.height,
          scaleOnDrag: j
        })
      }, m.key))
    })
  });
}
export {
  br as default
};
